proposals/001_interop.md

```diff
@@ -0,0 +1,464 @@
+# Docmaps Server API Protocol
+
+## Abstract
+
+This proposal is for a standard API contract for servers which offer Docmaps to interoperate with each other. This includes
+defining the endpoints that should be implemented for identifying the server, serving docmaps, searching for docmaps, including
+rules such as what endpoint results must be stable/cacheable and which may be dynamic.
+
+This proposes a server specification that conforms to W3C Linked Data Platform recommendations, because Docmaps data are
+natively considered to be Linked Data. Although an implementer of this protocol may not offer full linked data support,
+such as allowing `content-type: text/turtle`, and may not represent the data as a graph or triplestore in their persistent
+state solution, they are expected to implement this API at a minimum according to the usages of JSON-LD as described herein,
+in order to enable consumers of this data to seamlessly treat it as LD.
+
+## Motivation
+
+At this stage in the development of the Docmaps ecosystem, there is significant demand for Docmaps data for preprints but
+relatively little adoption by preprint repositories. Already various preprint repositories are implementing their own
+mechanisms for serving docmaps, which is encouraged. However, there is risk of divergent standards emerging that place
+significant burden on consumers to handle multiple API contracts. There is additional engineering overhead involved in
+building several different client and server libraries in parallel. With a majority of use cases covered by a single
+protocol, we can implement the protocol once per language and share the code; and clients written in various languages will
+naturally be interoperable.
+
+Additionally, beyond simply the vocabulary of endpoints and their responses, there is no consensus among current technical
+stakeholders as to what objects in a Docmaps data ecosystem are permanent or ephemeral, what should have integrity verification
+incorporated, and how to handle invalidation. In order to adequately describe the API contract for the initial Docmaps
+standard, this proposal answers several of these questions.
+
+A server that issues docmaps which conforms to this protocol will have an easier time acquiring users and consumers,
+supporting them, and interacting with other servers.
+
+## Goals
```

Comment:
Following conversation with COAR Notify, this server should be designed so that Notify payloads include links to this server's resources. That means
- endpoint that retrieving the history for a given document, possibly including a query parameter that speaks to a specific event or point in time -- i.e. deal directly with the IRIs of actions/events
- links with link-relations should be included

Reply:
core context https://www.eventnotifications.net/